client/my-sites/domains/domain-management/components/transfer-connected-domain-nudge/index.tsx

```diff
@@ -0,0 +1,43 @@
+import { Icon, starFilled } from '@wordpress/icons';
+import { useTranslate } from 'i18n-calypso';
+import moment from 'moment';
+import { useMyDomainInputMode } from 'calypso/components/domains/connect-domain-step/constants';
+import { domainUseMyDomain } from 'calypso/my-sites/domains/paths';
+import type { TransferConnectedDomainNudgeProps } from './types';
+import './style.scss';
+
+const TransferConnectedDomainNudge = ( {
+	siteSlug,
+	domain,
+}: TransferConnectedDomainNudgeProps ): JSX.Element | null => {
+	const translate = useTranslate();
+
+	const message = translate(
+		'Your domain is expiring soon. Consider {{a}}transferring it{{/a}} to WordPress.com to manage your site and domains all from one place.',
+		{
+			components: {
+				a: (
+					<a
+						href={ domainUseMyDomain( siteSlug, domain.name, useMyDomainInputMode.transferDomain ) }
+					/>
+				),
+			},
+		}
+	);
+
+	if (
+		! domain.registryExpiryDate ||
+		moment( domain.registryExpiryDate ).isAfter( moment().add( 45, 'days' ) )
+	) {
```

Comment:
Should we also check to make sure this date is in the future? Or at least a small number of days in the past too?

Reply:
Sounds good! The nudge is shown in the range of 45 days before expiration until 5 days after.